test/es6/generators-functionality.js

```diff
@@ -1766,6 +1766,23 @@ var tests = [
             );
         }
     },
+    {
+        name: "generator function returned through caller property",
+        body: function () {
+            var func = function () {
+                return func.caller;
+            }
+            function* gf(flag, value) {
+                if (!flag) {
+                    yield func();
+                }
+                yield value * value;
+            }
+            var callergf = gf().next().value;
+            assert.areEqual(callergf, gf, "Generator function returned through the caller property should be the same as the original generator function");
+            assert.areEqual(callergf(true, 10).next().value, 100, "Generator returned through the caller property should behave the same as the original generator function");

```

Comment:
Expected and actual are reversed in this assert.

Should be `assert.areEqual(100, callergf(...`

Same for the above.